### DISCUSSION TITLE
Is the amount of information about the tires of the main car and the control engine too little?

### DISCUSSION THREAD

**Discussion by Jackie2chen** [OpenSimulationInterface/open-simulation-interface#709] [Q&A]
Dear Author
     Is the amount of information about the tires of the main car and the control engine too little,in our control algorithm the information about the various directions of tire force speed acceleration and transmission output speed is essential,please ask if the information is not only in HostVehicleData.
                                                                                                                                 Thank you for your reply and I wish you good health.

**Comment by jdsika** [comment]
Hi 1823258216,

thank you for your question. Could you elaborate a litte bit on what exact information you are missing and where you would expect them to be? We are a group of many contributors who always welcome new ideas. 
The best way would be to create a pull request with your suggested changes. Try to make sure you do not break e.g. backwards compatibility if possible as it would make the extension easier. Also make sure to have as precisely described message as possible using units, self explaining names, define value ranges, put scientific references in as well.

Best regards
Carlo

**Comment by Jackie2chen** [comment]
Dear author:
In view of the lack of OSI content, we choose to extend the information we need in proto, and when transmitting it to other customers, they can take the existing data or extended data of OSI according to their needs, which is equivalent to a kind of compatibility. But I'm sorry that I can't disclose proto based on the company's internal simulation model because of the confidentiality agreement.
Wish you a happy life
Chen Hao

**Comment by jdsika** [comment]
Hi Chen,

I see, be aware that when interfacing with a partner or client the Mozilla Public License 2.0 states that you have to provide the source code of the interface you are using for the collaboration. So, if you add a field and compile the change to your custom version and you use this custom version with a partner of any sort you have to reveal the code to that partner. (At least this is up to MY knowledge - please read the license terms and consult company internally with the corresponding lawyers).
In this case of course a mutual NDA can state that this change is only visible and to be communicated between these partners. But if you extend the existing files than there is something like a "weal copy left" which means that all changes WITHIN a file are to be under the same license again (MPL 2.0). The extension of OSI in your case must be through additional private files and message extension.

But take this information with a grain of salt - you are talking to an engineer here and not a lawyer :)

My advise would be to ask your management if the disclossure of the information is more a benefit then a threat and that a visible contribution to the project might be the bigger gain!

Best regards
Carlo

**Reply by Jackie2chen** [reply]
Thank you for your suggestion and support. I will explain it to my leader
